dev/src/document.ts

```diff
@@ -349,7 +348,7 @@ export class DocumentSnapshot {
    *   console.log(`Retrieved data: ${JSON.stringify(data)}`);
    * });
    */
-  data(): DocumentData|undefined {
+  data(): {[field: string]: any}|undefined {  // tslint:disable-line no-any
```

Comment:
Are all of these remaining instances of "any" expected to stay forever? Or do you intend further cleanup passes?  In my ideal world we'd either remove them or comment them. :-)  This one may be intentional so that our users don't have to do instanceof checks on the data they get back?  I'm not sure though.

Reply:
> by using `any` as a return type we indicate that the user has to do some type checking of their own

Sorry for the nit, but that's not actually what `any` does. 😀  `any` makes it so you *don't* have to do any type checking (you can do anything you want with it and TypeScript won't complain).  If we want to force the user to do type-checking, then we should return `unknown` (new in TS 3.x) instead, which you can't do anything with unless you first do type-checking on it.

That said, using any here is consistent with the mobile SDK (https://github.com/firebase/firebase-js-sdk/blob/master/packages/firestore-types/index.d.ts#L24), so I'm fine with this.

Reply:
Uhm, that's what I meant (kind of, but not really).

Changed to
```
  // We deliberately use `any` in the external API to not impose type-checking
  // on end users.
```